net/hysteria/files/hysteria.init

```diff
@@ -0,0 +1,174 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=99
+
+package=$(basename $(readlink -f "$initscript"))
+prog="/usr/bin/hysteria"
+scripts_dir="/usr/share/${package}/bin"
+dnsmasq_confdir=$(uci -q  get dhcp.@dnsmasq[0].confdir || echo -n "/tmp/dnsmasq.d")
+
+_is_enabled() {
+	local instance="$1"; shift
+
+	local enabled
+	config_get_bool enabled "$instance" "enabled" "0"
+
+	[ "$enabled" == "1" ] || return 1
+}
+
+get_instance_data_value() {
+	local instance=$1; shift
+	local key=$1; shift
+
+	echo $(ubus call service list | jsonfilter -e "$.${package}.instances.${instance}.data.${key}")
+}
+
+setdown_tproxy() {
+	local instance=${1:-"*"}
+
+	local tproxy_list=$(get_instance_data_value "$instance" TPROXY)
+	local route_table_lits=$(get_instance_data_value "$instance" TPROXY_ROUTE_TABLE)
+	local table_list=$(get_instance_data_value "$instance" TPROXY_TABLE)
+
+	for i in $(seq $(echo "$tproxy_list" | wc -l)); do
+		local tproxy=$(get_bool $(echo "$tproxy_list" | sed -n "${i}p") 0)
+
+		[ "$tproxy" == "0" ] && continue
+
+		local table=$(echo "$table_list" | sed -n "${i}p")
+		local route_table=$(echo "$route_table_list" | sed -n "${i}p")
+
+		TABLE=${table} \
+		ROUTE_TABLE=${route_table} \
+			${scripts_dir}/set-tproxy down
+	done
+}
+
+clean_dnsmasq_conf() {
+	local instance=${1:-"*"}
+
+	find  $dnsmasq_confdir -type f -name "${package}_${instance}.conf" -exec rm {} \;
+}
+
+get_dnsmasq_conf() {
+	local instance=$1; shift
+	local server=$1; shift
+	local nftset=$1; shift
+
+	for domain; do
+		echo "server=/${domain}/${server}"
+		echo "nftset=/${domain}/${nftset}"
+	done
+}
+
+restart_dnsmasq() {
+	/etc/init.d/dnsmasq status 1>/dev/null && /etc/init.d/dnsmasq restart
```

Comment:
isn't `reload` a wiser option?

Reply:
重载不能保证加载新的配置，因为配置在配置文件中发生变化，并没有修改dnsmasq的实例配置
---
Reloading does not guarantee that the new configuration is loaded, because the configuration changes in the configuration file do not modify the instance configuration of dnsmasq.

Reply:
Do the dnsmasq features required need to use a separate dnsmasq instance, or config folder?

Reply:
The startup seems to indicate that only dismasq main config is used.

Reply:
我并不希望管理太多的dnsmasq相关的配置，仅仅是注入了一部分配置，或者可以选择注入配置的实例？

---
I don't want to manage too many dnsmasq related configurations, just inject part of the configuration, or can I choose the instance to inject the configuration?

Reply:
If you're using the main config, as you have in the script, then a reload should be suitable. But please test this.